cluster-autoscaler/processors/processors.go

```diff
@@ -67,6 +68,12 @@ type AutoscalingProcessors struct {
 	ActionableClusterProcessor actionablecluster.ActionableClusterProcessor
 	// ScaleDownCandidatesNotifier  is used to Update and Register new scale down candidates observer.
 	ScaleDownCandidatesNotifier *scaledowncandidates.ObserversList
+	// ScaleStateNotifier is used to register
```

Comment:
`ScaleStateNotifier.Notify(sddProcessor)` definitely reads better. Although I am worried it can misunderstood for a function which can be called on the same processor multiple times i.e., every time you want `ScaleStateNotifier` to notify a processor you call the `Notify` on it. Maybe `RegisterForNotifications` reads better?

Reply:
Renamed to `RegisterForNotifications` for now. Let me know what you think: [`e3930fc` (#5729)](https://github.com/kubernetes/autoscaler/pull/5729/commits/e3930fced86159cf4de62dcc0b553e0c419cda1b) (can change it again based on feedback)

Reply:
I think `Register` for adding a new observer is fine, my comment was regarding the line to which it is attached: 
```
// ScaleStateNotifier is used to register
```

I would expect it to say

```
// ScaleStateNotifier is used to notify about
```

Reply:
Updated the code.